### ISSUE TITLE
Request/response reply-to value

### ISSUE THREAD

**Issue by dmccarthy-bluefin** [basis-company/nats.php#17]
There are a number of issues with the value this lib uses as the reply-to value in request and dispatch functions. I want to discuss potential solutions with you before making any code changes as there are multiple ways to address the issues.

Here are the issues:

1. The prefix isn't consistent with other nats client libs or the nats docs (https://docs.nats.io/running-a-nats-service/configuration/securing_nats/authorization). This lib uses **inbox.** whereas they use **_INBOX.**. If a company using this lib is also using node or golang they would have to set duplicate authorization configurations for the same service.
2. The number of random characters used is small (e.g. inbox.5b08f327) and could cause collisions in larger deployments with lots of traffic.

Here are the potential changes that I want to make. Let me know which one you prefer or if you can think of an alternative:

1. Replace the existing code with `$sid = '_INBOX.' . bin2hex(random_bytes(16));`. The problem with this approach is that it could break existing implementations where the cluster permissions are based on the current value `inbox.>`.
2. Add a configuration setting to switch between the existing format and the format outlined above. Something like: replyToFormat = default|_INBOX_16 where default is used when no value is specified.
3. Add a configuration setting that takes a PHP function which is called when generating the id. With this approach the parent application can set whatever value they want in the function. Example:
```php
$configuration = new Configuration([
    'host' => 'localhost',
    'port' => 4222,
    'replyToGenerator' => function () {
        return '_INBOX.' . Cuid::cuid();
    },
]);
```

**Comment by adam-lmi** [comment]
I agree that we should be consistent with the other client implementations
I suggest to do the same as the JS client, and have an inboxPrefix parameter that is "_INBOX" by default: https://github.com/nats-io/nats.js/blob/41ef7c56c4ff4e163338e17f742575807735a702/README.md#connection-options
I also agree that we should extend the random part to at least 16 bytes. They use 22 bytes of base 36 ascii text for example: https://github.com/nats-io/node-nuid

Since we are still at 0.x version, I am not sure if we need a new major for this, and 0.8->0.9 could be enough

**Comment by dmccarthy-bluefin** [comment]
Agreed on the inboxPrefix approach. Should we also go with 22 byte base 36 ascii for the random part or are you happy with 16 byte hex?

**Comment by adam-lmi** [comment]
For the random part I think `bin2hex(random_bytes(16))` is okay. Fast running and easy to understand.
Implementing `Nuid` could be a nice feature, but then we should compare it's performance to bin2hex + random_bytes

**Comment by dmccarthy-bluefin** [comment]
Ok, I'm happy to move forward with this approach.

**Comment by adam-lmi** [comment]
Thank you for the contribution. I released it as 0.9.0
